os/emscripten/pre.js

```diff
@@ -30,32 +30,11 @@ Module.preRun.push(function() {
 
     Module.addRunDependency('syncfs');
     FS.syncfs(true, function (err) {
-        /* FS.mkdir() tends to fail if parent folders do not exist. */
-        if (!FS.analyzePath(content_download_dir).exists) {
-            FS.mkdir(content_download_dir);
-        }
-        if (!FS.analyzePath(content_download_dir + '/baseset').exists) {
-            FS.mkdir(content_download_dir + '/baseset');
-        }
-
-        /* Check if the OpenGFX baseset is already downloaded. */
-        if (!FS.analyzePath(content_download_dir + '/baseset/opengfx-0.6.0.tar').exists) {
-            window.openttd_downloaded_opengfx = true;
-            FS.createPreloadedFile(content_download_dir + '/baseset', 'opengfx-0.6.0.tar', 'https://binaries.openttd.org/installer/emscripten/opengfx-0.6.0.tar', true, true);
-        } else {
-            /* Fake dependency increase, so the counter is stable. */
-            Module.addRunDependency('opengfx');
-            Module.removeRunDependency('opengfx');
-        }
-
         Module.removeRunDependency('syncfs');
     });
 
     window.openttd_syncfs_shown_warning = false;
-    window.openttd_syncfs = function() {
-        /* Copy the virtual FS to the persistent storage. */
-        FS.syncfs(false, function (err) { });
-
+    window.openttd_syncfs_warn = function() {
```

Comment:
I think we should make this function call `FS.syncfs`, and take as parameter a callback to call when things are done.

This because there is now a lot of duplication going on, where there doesn't have to be.

We could even collapse this function with `openttd_syncfs`, and have the callback default to `undefined`, and not being called in that case. This means `openttd_exit` will look something like:

```
window.openttd_exit = function() {
  window.openttd_sync(() => {
    Module.onExit();
  });
}
```

Minus any typos, ofc.